jupyterhub/schema.yaml

```
@@ -212,6 +212,89 @@ properties:
           the `--values` or `-f` flag. During merging, lists are replaced while
           dictionaries are updated.
           ```
+      extraFiles: &extraFiles
+        type: string
+        description: |
+          A dictionary with extra files that should be injected into the pod's
+          container on startup.
+
+          ```yaml
+          hub:
+            extraFiles:
+              <implicit file name>:
+                # name is an optional way to override the implicit file name
+                # that must be unique.
+                name: <explicit file name>
+
+                # mode is by default 0644 and you can optionally override it
+                # either by octal notation (example: 0400) or decimal notation
+                # (example: 256).
+                mode: <file system permissions>
+
+                # mountDir is required and must be the directory you want to
+                # mount the file.
+                mountDir: <directory path>
+
+                # Choose one out of the three ways to represent the actual file
+                # content: data, stringData, or binaryData.
```

```suggestion
                # content: stringData, binaryData or data.
```
I think it helps to keep the ordering consistent with the following explanations.

---

I resolved this by updating the order below instead, I wanted to highlight the benefits of `data` so I let it be the first thing described.